src/test/scala/com/workday/warp/persistence/mysql/WarpSlickDslSpec.scala

```diff
@@ -48,12 +57,116 @@ class WarpSlickDslSpec extends WarpJUnitSpec with CorePersistenceAware {
     Thread.sleep(2000)
 
     val query1 = TestExecution.filter(_.endTime isWithinPast "1 SECOND")
+    println(query1.result.statements.head)
     this.persistenceUtils.runWithRetries(query1.result, 5) shouldBe empty
 
     val query2 = TestExecution.filter(_.endTime isWithinPast "5 MINUTE")
     this.persistenceUtils.runWithRetries(query2.result, 5).size shouldBe 3
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests YEAR dsl. */
+  def returnYear(): Unit = {
+    val testExecution: TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date, 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query1: Rep[Int] = timeStamp year()
+    this.persistenceUtils.runWithRetries(query1.result, 5) shouldBe Year.now.getValue
+
+    val query2: Query[Rep[Int], Int, Seq] = TestExecution.map(t => t.startTime year())
+    this.persistenceUtils.runWithRetries(query2.result, 5).head shouldBe Year.now.getValue
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests DATE dsl. */
+  def returnDate(): Unit = {
+    val format: SimpleDateFormat = new java.text.SimpleDateFormat("yyyy-MM-dd")
+    val date: String = format.format(new java.util.Date())
+
+    val testExecution: TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date, 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query1: Rep[String] = timeStamp date()
+    this.persistenceUtils.runWithRetries(query1.result, 5) shouldBe date
+
+    val query2: Query[Rep[String], String, Seq] = TestExecution.map(t => t.startTime date())
+    this.persistenceUtils.runWithRetries(query2.result, 5).head shouldBe date
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+  /** Tests NOW dsl. */
+  def getCurrentTimestamp(): Unit = {
+    val query: Rep[String] = TimeStampExtensions.now()
+    val result: String = this.persistenceUtils.runWithRetries(query.result, 5)
+
+    val UTCZone: ZoneId = ZoneId.of("UTC")
+    val zonedTime: ZonedDateTime = ZonedDateTime.now
+    val utcDate: ZonedDateTime = zonedTime.withZoneSameInstant(UTCZone)
+    val dateAsString: String = DateTimeFormatter.ofPattern("yyyy-MM-dd HH:mm:ss").format(utcDate)
+
+    result shouldBe dateAsString
+
+  }
+
+  @Test
+  @Category(Array(classOf[UnitTest]))
+   /** Tests UNIX_TIMESTAMP dsl. */
+  def returnUNIXTimeStamp(): Unit = {
+    val testExecution: TestExecutionRowLike = this.persistenceUtils.createTestExecution(methodSignature1, new Date(), 1.0, 10)
+    val timeStamp: Rep[Timestamp] = testExecution.startTime
+    val query: Rep[Int] = timeStamp UNIX_TIMESTAMP()
+    val result: Int = this.persistenceUtils.runWithRetries(query.result, 5)
+    val unixTimestamp: Long = Instant.now.getEpochSecond()
```

Comment:
hmm not sure if this test is very failsafe. If a second passes in runtime between lines 119 and 123 I think the test is going to fail.

Reply:
@sarcasticsimba  @tomnis is there a good way to test now? Since the time is constantly changing.

Reply:
Not sure if there's a way to account for the instant. Off the top of my head, a test we can perform is executing this code in a time block, and assert that `(unixTimestamp - result) shouldBe < elapsedTime)`. Also, I see that the `getEpochSecond` method returns a `Long`. Should we be returning a `Rep[Long]` instead?

Reply:
scalatest has `+-` syntax for checking against a range
```
seven should be (6 +- 2)
```
http://www.scalatest.org/user_guide/using_matchers#checkingNumbersAgainstARange